collector.go

```diff
@@ -28,19 +28,33 @@ const (
 var (
 	labelNames = []string{"ip", "host"}
 
+	pingResponseSeconds *prometheus.HistogramVec
+
+	hasRegistered bool
+)
+
+func init() {
+	hasRegistered = false
+}
+
+func setHistogramOptions(buckets []float64) {
```

Comment:
Rather than set a global variable, have a function like:

```go
func newPingResponseHistogram(buckets []float64) **prometheus.HistogramVec {
```

Then in main you can do this:
```go
prometheus.MustRegister(newPingResponseHistogram(buckets))
```